src/bin/pgcopydb/file_iterator.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * src/bin/pgcopydb/file_iterator.c
+ *   Implementations of a file iterator for reading new lıne seperated files
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "log.h"
+#include "file_iterator.h"
+#include "file_utils.h"
+
+/*
+ * A struct to hold the state of the file iterator. This
+ * iterator is used to iterate a file line-by-line.
+ */
+typedef struct FileIterator
+{
+	FILE *file;
+	const char *filename;
+	char *line;
+	size_t line_num;
+} FileIterator;
+
+/*
+ * Create a new iterator from the file name
+ */
+FileIterator *
+file_iterator_from(const char *filename)
+{
+	FileIterator *iterator = (FileIterator *) calloc(1, sizeof(FileIterator));
+	if (iterator == NULL)
+	{
+		log_error("Failed to allocate memory for FileIterator");
+		return NULL;
+	}
+
+	iterator->filename = filename;
+	iterator->file = fopen_read_only(filename);
+	if (iterator->file == NULL)
+	{
+		log_error("Failed to open file");
+		file_iterator_destroy(iterator);
+		return NULL;
+	}
+
+	return iterator;
+}
+
+
+/*
+ * Get the next line/item from the file/iterator.
+ *
+ * The memory for the line is allocated by this function and will be freed by it too.
+ */
+bool
+file_iterator_next(FileIterator *iterator, char **line)
+{
+	*line = NULL;
+	if (iterator->line)
+	{
+		free(iterator->line);
+		iterator->line = NULL;
+	}
+	size_t len = 0;
+	if (getline(&(iterator->line), &len, iterator->file) != -1)
```

Comment:
Note to myself: compare getline() and fgets()

Reply:
So `getline()` implements its own memory allocation, which looks nice, but I wonder if that works well with libgc, the Garbage Collector that we use. It might be easier for us to use `fgets()`.